test/loader/serializeTypeTest.cpp

```diff
@@ -275,4 +275,74 @@ TEST(serializeTypeTest, SerializeGlobalType) {
   EXPECT_FALSE(
       SerNoRefType.serializeSection(createGlobalSec(GlobalType), Output));
 }
+
+TEST(serializeTypeTest, SerializeFuncRefType) {
+  std::vector<uint8_t> Output;
+  std::vector<uint8_t> Expected;
+
+  WasmEdge::AST::GlobalType GlobalType;
+
+  // 1. test FuncRef
+  GlobalType.setValType(WasmEdge::TypeCode::FuncRef);
+  GlobalType.setValMut(WasmEdge::ValMut::Const);
+  Output = {};
+
+  EXPECT_TRUE(Ser.serializeSection(createGlobalSec(GlobalType), Output));
+  Expected = {
+      0x06U, // Global section
+      0x04U, // Content size = 4
+      0x01U, // Vector length = 1
+      0x70U, // FuncRef type
+      0x00U, // Const mutation
+      0x0BU  // Expression
+  };
+  EXPECT_EQ(Output, Expected);
+
+  // 2. test ExternRef
+  GlobalType.setValType(WasmEdge::TypeCode::ExternRef);
+  GlobalType.setValMut(WasmEdge::ValMut::Const);
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createGlobalSec(GlobalType), Output));
+  Expected = {
+      0x06U, // Global section
+      0x04U, // Content size = 4
+      0x01U, // Vector length = 1
+      0x6FU, // ExternRef type
+      0x00U, // Const mutation
+      0x0BU  // Expression
+  };
+  EXPECT_EQ(Output, Expected);
+
+  // Why does the following test give Segmentation fault
```

Comment:
It's not segmentation fault. It ran into unreachable.
You should realize the constructors of `class ValType` first.